admin/server/handlers/cosmosdump/dump_cosmos.go

```diff
@@ -38,46 +37,12 @@ func (h *CosmosDumpHandler) ServeHTTP(w http.ResponseWriter, req *http.Request)
 
 func (h *CosmosDumpHandler) serveHTTP(w http.ResponseWriter, request *http.Request) error {
 	ctx := request.Context()
-	logger := utils.LoggerFromContext(ctx)
 
 	// get the azure resource ID for this HCP
 	resourceID, err := utils.ResourceIDFromContext(request.Context())
 	if err != nil {
 		return utils.TrackError(err)
 	}
 
-	// load the HCP from the cosmos DB
-	cosmosCRUD, err := h.cosmosClient.UntypedCRUD(*resourceID)
-	if err != nil {
-		return utils.TrackError(err)
-	}
-	startingCosmosRecord, err := cosmosCRUD.Get(ctx, resourceID)
-	if err != nil {
-		return utils.TrackError(err)
-	}
-	allCosmosRecords, err := cosmosCRUD.ListRecursive(ctx, nil)
-	if err != nil {
-		return utils.TrackError(err)
-	}
-
-	errs := []error{}
-	content, err := json.Marshal(startingCosmosRecord)
-	if err != nil {
-		errs = append(errs, err)
-	}
-	logger.Info(string(content))
-
-	for _, typedDocument := range allCosmosRecords.Items(ctx) {
-		content, err := json.Marshal(typedDocument)
-		if err != nil {
-			errs = append(errs, err)
-		}
-		logger.Info(string(content))
-	}
-	if err := allCosmosRecords.GetError(); err != nil {
-		return utils.TrackError(err)
-	}
-
-	return utils.TrackError(errors.Join(errs...))
-
+	return serverutils.DumpDataToLogger(ctx, h.cosmosClient, resourceID)
```

Comment:
eventually we might want the dump to be returned as payload here. logging and just reporting errors is fine for now